File: 1646-kth-missing-positive-number/kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int cnt = 0, ans = 1,i = 0;
        while (cnt < k) {
            if (i < arr.size() and arr[i] == ans) {
                i++; 
            } else {
                cnt++; 
                if (cnt == k)
                    return ans;
            }
            ans++;
        }
        return 0; 
    }
};

File: 1646-kth-missing-positive-number/kth-missing-positive-number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    let cnt = 0, ans = 1, i = 0;
    while(cnt < k){
        if(i < arr.length && arr[i] == ans){
            i++;
        } 
        else{
            cnt++;
            if(cnt == k)return ans;
        }
        ans++;
    }
    return 0;
};